sound/soc/intel/common/soc-acpi-intel-arl-match.c

```diff
@@ -44,6 +44,31 @@ static const struct snd_soc_acpi_endpoint spk_3_endpoint = {
 	.group_id = 1,
 };
 
+/*
+ * RT722 is a multi-function codec, three endpoints are created for
+ * its headset, amp and dmic functions.
+ */
+static const struct snd_soc_acpi_endpoint rt722_endpoints[] = {
+	{
+		.num = 0,
+		.aggregated = 0,
+		.group_position = 0,
+		.group_id = 0,
+	},
+	{
+		.num = 1,
+		.aggregated = 0,
+		.group_position = 0,
+		.group_id = 0,
+	},
+	{
+		.num = 2,
+		.aggregated = 0,
+		.group_position = 0,
+		.group_id = 0,
+	},
```

Comment:
We need to remove the Speaker endpoint if it is not used.

Reply:
@plbossart Can you take a look at https://github.com/thesofproject/linux/pull/5188?

Reply:
@charleskeepax has a good suggestion on https://github.com/thesofproject/linux/pull/5186 that can make a DAI be optional and present by default.

Reply:
> @charleskeepax has a good suggestion on #5186 that can make a DAI be optional and present by default.

@bardliao, thank you for your advice. By referencing @charleskeepax #5186 suggestion on cs42l43, I could apply the quirk_exclude codes into RT722 as a short-term once #5186 merged. In the end, #5188 implementing the "endpoints present by sdca acpi function" will be a final solution afterwards.